backend/inspirehep/editor/cache.py

```diff
@@ -0,0 +1,89 @@
+from datetime import datetime
+
+import structlog
+from flask import current_app
+from redis import StrictRedis
+
+from inspirehep.config import REDIS_KEY_VALUE_PAIR_EXPIRE_TIME
+
+LOGGER = structlog.getLogger()
+
+
+class EditorCache(object):
+    def __init__(
+        self,
+        recid,
+        record_version,
+        user_email=None,
+        task_name=None,
+    ):
+        self.recid = recid
+        self.record_version = record_version
+        self.user_email = user_email
+        self.task_name = task_name
+
+    @property
+    def redis(self):
+        redis_url = current_app.config.get("CACHE_REDIS_URL")
+        redis = StrictRedis.from_url(redis_url)
+        return redis
+
+    @property
+    def user_hash_name(self):
+        return f"editor-lock:{self.recid}@{self.record_version}"
+
+    @property
+    def task_hash_name(self):
+        return f"editor-task-lock:{self.recid}@{self.record_version}"
+
+    @property
+    def user_lock_message(self):
+        return f"{self.user_email} at {datetime.now().isoformat()}"
+
+    @property
+    def task_lock_message(self):
+        return f"{self.task_name} at {datetime.now().isoformat()}"
+
+    def add_user_lock(self):
+        self.redis.hset(
+            self.user_hash_name, key=self.user_email, value=self.user_lock_message
+        )
+        self.redis.expire(self.user_hash_name, time=REDIS_KEY_VALUE_PAIR_EXPIRE_TIME)
+
+    def add_task_lock(self):
+        self.redis.hset(
+            self.task_hash_name, key=self.task_name, value=self.task_lock_message
+        )
+        self.redis.expire(self.task_hash_name, time=REDIS_KEY_VALUE_PAIR_EXPIRE_TIME)
+
+    def remove_task_lock(self):
+        self.redis.hdel(self.task_hash_name, self.task_name)
+
+    def remove_user_lock(self):
+        self.redis.hdel(self.user_hash_name, self.user_email)
+
```

Comment:
do you really need to have two copies of every method here? I would think you can detect which case you're in based on the data passed to the `__init__` (`task_name` or `user_email`). Alternatively, you could subclass a base class and have a task version and a user version. Currently the API doesn't make much sense. What happens if you call `add_task_lock` for a task initialized with a `user_email`  but no `task_name`.